crawl-ref/source/scroller.cc

```diff
@@ -176,11 +178,23 @@ bool formatted_scroller::process_key(int ch)
     }
 }
 
-void formatted_scroller::set_scroll(int y)
+// y: line to appear at top if the document is long enough. Go to bottom if not.
+// use_shade: Allow the top line to be partly shaded, where relevant.
+void formatted_scroller::scroll_to_line(int y, bool use_shade)
 {
+#ifdef USE_TILE_LOCAL
+    y *= tiles.get_crt_font()->char_height();
+    if (!use_shade)
+        y -= UI_SCROLLER_SHADE_SIZE/2; // Skip the shaded border at the top.
+#elif defined(USE_TILE_WEB)
+    if (!use_shade && y)
+        y--; // 1 line should be enough.
     if (from_webtiles)
         m_scroller->set_scroll(y);
     else
+#else
+    (void)use_shade;
+#endif
```

Comment:
Seems maybe marginally nicer to put this conditional earlier. Something like:

```
#ifdef USE_TILE_LOCAL
void formatted_scroller::scroll_to_line(int y, bool use_shade)
#else
void formatted_scroller::scroll_to_line(int y, bool /*use_shade*/)
#endif
```

But also I think we can probably just cut this `#else` - do we really have a linter that complains if we don't use a function parameter?

Reply:
1. It would show that it _is_ unused, 

On the other hand, it would need an extra proprocessor condition as it has two alternatives rather than three. That takes  the proportion of the code which comprises preprocessor stuff from 20% to 27%, which to me just makes it harder to read.

Another way to remove that line would be to have three versions of the whole scroll_to_line() function  They only share the last two code lines, so it may not need an extra function to handle those.

2. Yes, `-Wextra` brings in `-Wmissing-declarations` on G++.

I haven't changed anything for now.

Reply:
Ah, I hadn't spotted the intervening `#elif defined(USE_TILE_WEB)` - that complicates this for sure. Yeah, three separate versions of the function could be reasonable. I'd try it out and see how it looks?

Reply:
I've changed this now.